File: backend/src/main/java/com/ntg/sms/Entities/Dtos/Response/AttendanceDailyResponse.java
package com.ntg.sms.Entities.Dtos.Response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AttendanceDailyResponse {
    private LocalDate date;
    private long totalSessions;
    private long present;
    private long absent;
    private long late;
    private double attendancePercentage;
    private List<AttendaceResponse> attendanceList;
}

File: Frontend/Student/src/app/models/AttendanceDailyResponse.ts
// Matches AttendaceResponse.java exactly
export interface AttendanceResponse {
  id: number;
  studentId: number;
  studentName: string;
  sessionId: number | null;
  sessionNumber: number;       // slot position 1-8 in the day
  status: 'P' | 'A' | 'L';
  dateTime: string;
  courseName: string | null;   // Session → Course
  teacherName: string | null;  // Session → Course → Teacher → User
}

// Matches AttendanceDailyResponse.java exactly
export interface AttendanceDailyResponse {
  date: string;
  totalSessions: number;
  present: number;
  absent: number;
  late: number;
  attendancePercentage: number;
  attendanceList: AttendanceResponse[];
}